### ISSUE TITLE
Stylesheets not updating in middleman v4

### ISSUE THREAD

**Issue by danielmag** [middleman/middleman#1631]
Stylesheet partials aren't being updated when making changes.
If I change them to normal stylesheets (remove the underscore), everything works well

**Comment by tdreyno** [comment]
Are you on a RC gem or directly on github master?

**Comment by danielmag** [comment]
rc1

**Comment by tdreyno** [comment]
Can you double-check on master?

**Comment by danielmag** [comment]
i can try monday

**Comment by danielmag** [comment]
I have the same issue on master

**Comment by tdreyno** [comment]
Great, I'll dig in. To confirm, this is only related to the partials. Can you post your `Gemfile.lock`?

**Comment by johnnyshields** [comment]
:+1: seeing the same issue. Thanks!

**Comment by danielmag** [comment]
Don't think it's relevant. I'm seeing the same issue on the default middleman 4 template

**Comment by johnnyshields** [comment]
We're getting reports that JS files are also not being watched. This is a big annoyance for us, please help investigate soon.

**Comment by johnnyshields** [comment]
@tdreyno can you please help with this?

**Comment by sn3p** [comment]
Same problem (v4.0.0.rc.1)

**Comment by johnnyshields** [comment]
Pinging on this issue, quite difficult to use v4 without this fixed.

**Comment by johnnyshields** [comment]
I've investigated this and found the problem.

In `Middleman::SourceWatcher`, there's this code:

```
# line 235-ish
    def update(updated_paths, removed_paths)
      valid_updates = updated_paths
                      .lazy
                      .map(&method(:path_to_source_file))
                      .select(&method(:valid?))
                      .to_a
                      .each do |f|
                        add_file_to_cache(f)
                        logger.debug "== Change (#{f[:types].inspect}): #{f[:relative_path]}"
                      end
```

What's happening is that when you add/modify/remove a partial, it's busting the cache for the partial (e.g. `stylesheets/_variables.scss`) but not the .scss file in which it's included.

My understanding of scss nomenclature is that `_xxx.scss` (leading underscore) can only be included in other SCSS files, while `yyy.scss` (no leading underscore) can be rendered as either a standalone OR included in another SCSS file. Therefore, I think it makes sense that modifying any SCSS file (`_` or not) should bust the cache for all SCSS files which do not have a `_`. @tdreyno your thoughts?

(By the way, this problem is likely to apply to other types of assets).

**Comment by johnnyshields** [comment]
PR to fix this issue is here: #1695
